### PR TITLE
July 2023 Field Trip Fixes

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

This PR repairs a few field trip regressions as a result of recent merges
 
- Highlight works again
- Date time fields correct width
- Z-index of banned routes field
- Highlight external location field changes

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]
Changes look good, however they will need the corresponding OTP version that has the new fares data types before this PR should be merged. If not changing their OTP version, https://github.com/opentripplanner/otp-react-redux/pull/942/commits/216c480236f9137a1b04d653f2a411df44d3aadb should be used instead.

File: lib/actions/field-trip.js [hunk 1]
```diff
@@ -695,6 +695,7 @@ function checkValidityAndCapacity(state, request) {
       itinerary.legs
         .filter((leg) => isTransit(leg.mode))
         .forEach((leg) => {
+          // TODO: does this exist with OTP2?
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/actions/field-trip.js]: Thanks for figuring this out!

**Comment by miles-grant-ibigroup** [comment]
OTP-RR always targets the latest OTP2 version.

File: lib/actions/field-trip.js [hunk 2]
```diff
@@ -695,7 +695,9 @@ function checkValidityAndCapacity(state, request) {
       itinerary.legs
         .filter((leg) => isTransit(leg.mode))
         .forEach((leg) => {
-          const { tripId } = leg
+          let { tripId } = leg
+          // OTP2 support
+          if (!tripId) tripId = leg?.trip?.id
```
Comment (daniel-heppner-ibigroup) [inline lib/actions/field-trip.js:700]: Just use the OTP2 mode always! OTP1 support is gone so there's no point in keeping legacy stuff around.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Looks good, thanks for the touch ups

**Pull request merged**